x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/FieldAttribute.java

```diff
@@ -242,4 +243,14 @@ protected String label() {
     public EsField field() {
         return field;
     }
+
+    /**
+     * If this field is unsupported this contains the underlying ES types. If there
+     * is a type conflict this will have many elements, some or all of which may
+     * be actually supported types.
+     */
+    @Nullable
+    public List<String> originalTypes() {
+        return null;
+    }
```

Comment:
If we're going to move it, maybe just put it in `UnsupportedAttribute`. I don't like that either, but it feels better than this in the middle spot.

Reply:
I think we wanted to keep it on `FieldAttribute` level as we might want to create one (or subtype) from a `MultiTypeEsField` at some point. But anyways I pushed it further down now and we can revise this if/when needed.